From 3a93877113c15ddc844a9fdb08757feea4aa4c68 Mon Sep 17 00:00:00 2001
From: Sindhu S <sindhus@live.in>
Date: Mon, 30 Dec 2013 12:42:20 +0530
Subject: [PATCH] Gnu Screen Vertical Split patch

The patch from the original author doesn't apply cleanly and so I
had to edit the source files manually to add the code from the diff:
http://vsp4sdl.yuggoth.org/wrp_vertical_split_0.3_4.0.2.diff.bz2.

Homepage for the vertical split code: http://fungi.yuggoth.org/vsp4s/.
---
 src/ansi.c   |  2 +-
 src/pty.c    |  1 +
 src/resize.c | 15 +++++++++++++--
 3 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/src/ansi.c b/src/ansi.c
index 04c9012..99122fd 100644
--- a/src/ansi.c
+++ b/src/ansi.c
@@ -568,7 +568,7 @@ register int len;
 		{
 		case '0': case '1': case '2': case '3': case '4':
 		case '5': case '6': case '7': case '8': case '9':
-		  if (curr->w_NumArgs < MAXARGS)
+			if (curr->w_NumArgs >= 0 && curr->w_NumArgs < MAXARGS)
 		    {
 		      if (curr->w_args[curr->w_NumArgs] < 100000000)
 			curr->w_args[curr->w_NumArgs] =
diff --git a/src/pty.c b/src/pty.c
index def7f93..83a4ec0 100644
--- a/src/pty.c
+++ b/src/pty.c
@@ -39,6 +39,7 @@
 #endif
 
 /* for solaris 2.1, Unixware (SVR4.2) and possibly others */
+#if defined(HAVE_SVR4_PTYS) && !defined(__APPLE__)
 #ifdef HAVE_STROPTS_H
 # include <sys/stropts.h>
 #endif
diff --git a/src/resize.c b/src/resize.c
index 362fc9a..6c1ff0d 100644
--- a/src/resize.c
+++ b/src/resize.c
@@ -663,6 +663,17 @@ int wi, he, hi;
   if (wi <= 0 || he <= 0)
     wi = he = hi = 0;
 
+    if (wi > 1000)
+    {
+      Msg(0, "Window width too large, truncated");
+      wi = 1000;
+    }
+  if (he > 1000)
+    {
+      Msg(0, "Window height too large, truncated");
+      he = 1000;
+    }
+
   if (p->w_type == W_TYPE_GROUP)
     return 0;
 
@@ -954,8 +965,8 @@ int wi, he, hi;
 	      return -1;
 	    }
 	  for (; t < wi; t++)
-	    p->w_tabs[t] = t && !(t & 7) ? 1 : 0; 
-	  p->w_tabs[wi] = 0; 
+     p->w_tabs[t] = t && !(t & 7) ? 1 : 0;
+   p->w_tabs[wi] = 0;
 	}
       else
 	{
-- 
1.8.5.2

